wallet/createtx.go

```diff
@@ -380,122 +207,17 @@ func (w *Wallet) findEligibleOutputs(account uint32, minconf int32, bs *waddrmgr
 	return eligible, nil
 }
 
-// signMsgTx sets the SignatureScript for every item in msgtx.TxIn.
-// It must be called every time a msgtx is changed.
-// Only P2PKH outputs are supported at this point.
-func signMsgTx(msgtx *wire.MsgTx, prevOutputs []wtxmgr.Credit, mgr *waddrmgr.Manager, chainParams *chaincfg.Params) error {
-	if len(prevOutputs) != len(msgtx.TxIn) {
-		return fmt.Errorf(
-			"Number of prevOutputs (%d) does not match number of tx inputs (%d)",
-			len(prevOutputs), len(msgtx.TxIn))
-	}
-	for i, output := range prevOutputs {
-		// Errors don't matter here, as we only consider the
-		// case where len(addrs) == 1.
-		_, addrs, _, _ := txscript.ExtractPkScriptAddrs(output.PkScript,
-			chainParams)
-		if len(addrs) != 1 {
-			continue
-		}
-		apkh, ok := addrs[0].(*btcutil.AddressPubKeyHash)
-		if !ok {
-			return ErrUnsupportedTransactionType
-		}
-
-		ai, err := mgr.Address(apkh)
-		if err != nil {
-			return fmt.Errorf("cannot get address info: %v", err)
-		}
-
-		pka := ai.(waddrmgr.ManagedPubKeyAddress)
-		privkey, err := pka.PrivKey()
-		if err != nil {
-			return fmt.Errorf("cannot get private key: %v", err)
-		}
-
-		sigscript, err := txscript.SignatureScript(msgtx, i,
-			output.PkScript, txscript.SigHashAll, privkey,
-			ai.Compressed())
-		if err != nil {
-			return fmt.Errorf("cannot create sigscript: %s", err)
-		}
-		msgtx.TxIn[i].SignatureScript = sigscript
-	}
-
-	return nil
-}
-
-func validateMsgTx(msgtx *wire.MsgTx, prevOutputs []wtxmgr.Credit) error {
-	for i := range msgtx.TxIn {
-		vm, err := txscript.NewEngine(prevOutputs[i].PkScript,
-			msgtx, i, txscript.StandardVerifyFlags, nil)
+func validateMsgTx(tx *wire.MsgTx, prevScripts [][]byte) error {
```

Comment:
Needs a comment.  Particularly about how the order of the scripts here has to match up with the order they are consumed by the tx inputs.  That bit is important because it means the caller has to ensure it actually reorders then correctly if it randomizes the transaction's inputs.